pkg/sfu/downtrack.go

```diff
@@ -1322,14 +1329,49 @@ func (d *DownTrack) handleRTCP(bytes []byte) {
 			for _, pair := range p.Nacks {
 				nackedPackets = append(nackedPackets, d.sequencer.getSeqNoPairs(pair.PacketList())...)
 			}
-			if err = d.receiver.RetransmitPackets(d, nackedPackets); err != nil {
-				return
-			}
+			go d.retransmitPackets(nackedPackets)
 		}
 	}
+}
 
-	if len(fwdPkts) > 0 {
-		d.receiver.SendRTCP(fwdPkts)
+func (d *DownTrack) retransmitPackets(nackedPackets []packetMeta) {
+	src := packetFactory.Get().(*[]byte)
+	defer packetFactory.Put(src)
+	for _, meta := range nackedPackets {
+		pktBuff := *src
+		n, err := d.receiver.ReadSimulcastRTP(pktBuff, meta.layer, meta.sourceSeqNo)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			continue
+		}
+		var pkt rtp.Packet
+		if err = pkt.Unmarshal(pktBuff[:n]); err != nil {
+			continue
+		}
+		pkt.Header.SequenceNumber = meta.targetSeqNo
+		pkt.Header.Timestamp = meta.timestamp
+		pkt.Header.SSRC = d.ssrc
+		pkt.Header.PayloadType = d.payloadType
+
+		err = d.MaybeTranslateVP8(&pkt, meta)
```

Comment:
I think both `MaybeTranslateVP8` and `WriteHeaderExtension` below can be made private (guess it could have been made private even before as they were in the same package).

Reply:
Done